stactools_threedep/stactools/threedep/commands.py

```diff
@@ -0,0 +1,126 @@
+                                          USGS_FTP_BASE, USGS_3DEP_ID)
+from stactools.threedep import utils, stac
+
+
+def create_threedep_command(cli):
+    """Creates the threedep command line utility."""
+    @cli.group("threedep", short_help="Work with USGS 3DEP elevation data.")
+    def threedep():
+        pass
+
+    @threedep.command(
+        "create-collection",
+        short_help="Create a STAC collection for existing USGS 3DEP data")
+    @click.argument("destination")
+    @click.option("-s",
+                  "--source",
+                  help="The href of a directory tree containing metadata",
+                  default=None)
+    @click.option("-i",
+                  "--id",
+                  multiple=True,
+                  help="Ids to fetch. If not provided, will fetch all IDs.")
+    @click.option("--quiet/--no-quiet", default=False)
+    def create_collection_command(destination, source, id, quiet):
+        """Creates a 3DEP collection in DESTINATION.
+
+        If SOURCE is not provided, will use the metadata in AWS. SOURCE is
+        expected to be a directory tree mirroring the structure on USGS, so
+        it is best created using `stac threedep download-metadata`.
+        """
+        base_ids = id  # not sure how to rename arguments in click
+        items = {}
+        for product in PRODUCTS:
+            items[product] = []
+            if base_ids:
+                ids = base_ids
+            else:
+                ids = utils.fetch_ids(product)
+            for id in ids:
+                item = stac.create_item_from_product_and_id(
+                    product, id, source)
+                items[product].append(item)
+                if not quiet:
+                    print(item.id)
+        all_items = [item for sublist in items.values() for item in sublist]
+        if not all_items:
+            raise Exception("no items found")
+        extent = Extent.from_items(all_items)
+        collection = Collection(
+            id=USGS_3DEP_ID,
+            title="USGS 3DEP DEMs",
+            keywords=["USGS", "3DEP", "NED", "DEM", "elevation"],
+            providers=[USGS_PROVIDER],
+            description=DESCRIPTION,
+            extent=extent,
+            license="PDDL-1.0")
+        for product in PRODUCTS:
+            if product == "1":
+                title = "1 arc-second"
+                description = "USGS 3DEP 1 arc-second DEMs"
+            elif product == "13":
+                title = "1/3 arc-second"
+                description = "USGS 3DEP 1/3 arc-second DEMs"
+            else:
+                raise NotImplementedError
+            catalog = Catalog(id=product, description=description, title=title)
+            collection.add_child(catalog)
+            catalog.add_items(items[product])
+        collection.update_extent_from_items()
+        collection.normalize_hrefs(destination)
+        collection.save(catalog_type=CatalogType.SELF_CONTAINED)
```

Comment:
With the recent change in STAC spec, Collection does not inherit from Catalog, which means that a Collection cannot act as a root catalog.
It looks like here a top-level Collection is created and saved without a root. 

I'm not sure if this is common across other stactools modules, but I think the workflow should for creating any new collection should first assume an existing root catalog, and new collections are added to that.

Reply:
![Screen Shot 2021-04-05 at 2 30 51 PM](https://user-images.githubusercontent.com/58314/113623402-897ebd00-961b-11eb-9b62-55bf8d166c10.png)

FYI I think my confusion arose from [here](https://github.com/radiantearth/stac-spec/blob/master/catalog-spec/catalog-spec.md#link-object), where it says that `root` can point to a `Catalog` or a `Collection`.

I will still fix to implement `Catalog -> 2xCollection (one for 1 arc-second and one for 1/3 arc second -> Items`, with maybe some sort of subsetting catalogs to group the items better as well.

Reply:
Ah, good catch.  That should be updated as a consequence of not having Collections inherit from Catalogs.
However in looking through the latest stac-spec version I'm having trouble finding the changes that were made to that wording.  Will open up issue

Reply:
I'm still not convinced that there should be two collections here, vs one collection where you would be able to filter on `gsd`.  We should get some additional feedback on that.

Reply:
Roger. I guess it feels weird to me then to have a `1 -> 1 -> many` tree -- feels like there's an unnecessary node.

Reply:
You mean if it was Catalog->Collection->Items?
You'd normally have other non-3DEP collections in your Catalog, it also mirrors a STAC API where there's a root catalog with Collections under it.

Reply:
Gotcha. Yeah, I guess I'm thinking about if we expand this function to create a `Catalog` that encompasses more than just the 1 and 1/3 arc second data, e.g. the 1m which has less complete coverage. In that case, my thinking is we'd want a separate `Collection` for the 1m at least, so that its bounds are tighter than the 1 arc-second. Or am I overusing `Collection`s?